proposals/more-const-types.md

```diff
@@ -0,0 +1,79 @@
+# Unmanaged constant types
```

Comment:
I have a similar proposal, from 2017, that goes over a different approach to supporting this functionality here: https://github.com/dotnet/csharplang/discussions/688